verification/api/challengeresponsesession.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2022 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+)
+
+type Status uint8
+
+const (
+	StatusWaiting Status = iota
+	StatusProcessing
+	StatusComplete
+	StatusFailed
+)
+
+func (o Status) String() string {
+	switch o {
+	case StatusWaiting:
+		return "waiting"
+	case StatusProcessing:
+		return "processing"
+	case StatusComplete:
+		return "complete"
+	case StatusFailed:
+		return "failed"
+	}
+	return "unknown"
+}
+
+func (o *Status) FromString(s string) error {
+	switch s {
+	case "waiting":
+		*o = StatusWaiting
+	case "processing":
+		*o = StatusProcessing
+	case "complete":
+		*o = StatusComplete
+	case "failed":
+		*o = StatusFailed
+	default:
+		return fmt.Errorf("unknown status %s", s)
+	}
+	return nil
+}
+
+func (o Status) MarshalJSON() ([]byte, error) {
+	return json.Marshal(o.String())
+}
+
+func (o *Status) UnmarshalJSON(b []byte) error {
+	var s string
+
+	if err := json.Unmarshal(b, &s); err != nil {
+		return err
+	}
+
+	return o.FromString(s)
+}
+
+type Blob struct {
+	Type  string `json:"type"`
+	Value []byte `json:"value"`
+}
+
+type ChallengeResponseSession struct {
+	id            string
+	Status        Status    `json:"status"`
+	Nonce         []byte    `json:"nonce"`
+	Expiry        time.Time `json:"expiry"`
+	Accept        []string  `json:"accept"`
+	Evidence      *Blob     `json:"evidence,omitempty"`
+	Result        *[]byte   `json:"result,omitempty"`
```

Comment:
Could line 76: Result be named as AttestationResult, Otherwise it is very confusing to have 
(a) Status (b) Result (c) Failure Reason ?

Reply:
this also is a 1:1 map of the OpenAPI description: https://github.com/veraison/docs/blob/main/api/challenge-response/schemas/components.yaml

Reply:
Correct: All I am saying that change Result in go field to Attestation Result:
========================

          $ref: '#/components/schemas/EvidenceBlob'
          description: >
            the submitted attestation Evidence
        result:
          $ref: '#/components/schemas/AttestationResult'

============================
Also where is the AttestationResult enumerated?  As per the API documentation?

Reply:
> Correct: All I am saying that change Result in go field to Attestation Result:

it's called `result` in the API, why should we deviate?

> Also where is the AttestationResult enumerated? As per the API documentation?

the format of the result is determined by the Verification module, the API frontend will forward whatever it's given.